zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/SparkSQLGenerator.scala

```diff
@@ -0,0 +1,636 @@
+    }.mkString("\n")
+    appendIfNotEmpty(queryBuilder, lookupStr)
+
+    ParsedResult(Seq.empty, (producedExprs).result().toMap)
+  }
+
+
+
+
+
+  protected def parseFilterRelation(filter: FilterStatement, fieldExpr: String): String = {
+    filter.field.dataType match {
+      case DataType.Number =>
+        parseNumberRelation(filter, fieldExpr)
+      case DataType.Time =>
+        parseTimeRelation(filter, fieldExpr)
+      //      case DataType.Point =>
+      //        parsePointRelation(filter, fieldExpr)
+      case DataType.Boolean => ???
+      case DataType.String =>
+        parseStringRelation(filter, fieldExpr)
+      case DataType.Text =>
+        parseTextRelation(filter, fieldExpr)
+      case DataType.Bag => ???
+      case DataType.Hierarchy =>
+        throw new QueryParsingException("the Hierarchy type doesn't support any relations.")
+      case _ => throw new QueryParsingException(s"unknown datatype: ${filter.field.dataType}")
+    }
+  }
+
+  protected def parseTimeRelation(filter: FilterStatement,
+                                  fieldExpr: String): String = {
+    filter.relation match {
+      case Relation.inRange => {
+        s"$fieldExpr >= '${filter.values(0)}' and $fieldExpr < '${filter.values(1)}'"
+      }
+      case _ => {
+        s"$fieldExpr ${filter.relation} '${filter.values(0)}'"
+      }
+    }
+  }
+
+  protected def parseStringRelation(filter: FilterStatement, fieldExpr: String): String = {
+    filter.relation match {
+      case Relation.matches => {
+        val values = filter.values.map(_.asInstanceOf[String])
+        s"""$fieldExpr="${values(0)}""""
+      }
+      case Relation.!= => {
+        val values = filter.values.map(_.asInstanceOf[String])
+        s"""$fieldExpr!="${values(0)}""""
+      }
+      case Relation.contains => ???
+
+    }
+  }
+  protected def parseTextRelation(filter: FilterStatement, fieldExpr: String): String = {
+    val wordsArr = ArrayBuffer[String]()
+    filter.values.foreach(w => wordsArr += "%" + w + "%")
+    val sb = new StringBuilder
+    for (i <- 0 until (wordsArr.length - 1)){
+      sb.append(s"lower($fieldExpr) ${fullTextContains} '${wordsArr(i)}' and ")
+    }
+    sb.append(s"lower($fieldExpr) ${fullTextContains} '${wordsArr(wordsArr.length - 1)}'")
+    sb.toString()
+  }
+
+
+  protected def parseGeoCell(scale: Double, fieldExpr: String, dataType: DataType.Value): String = {
+    val origin = s"${createPoint}(0.0,0.0)"
+    s"${getPoints}(${spatialCell}(${fieldExpr}, $origin, ${1 / scale}, ${1 / scale}))[0]"
+  }
+
+  protected def parseAggregateFunc(aggr: AggregateStatement,
+                                   fieldExpr: String): String = {
+    def aggFuncExpr(aggFunc: String): String = {
+      if (aggr.field.name.equals("*")) {
+        s"$aggFunc($groupVar)"
+      } else {
+        s"$aggFunc( (select value $groupVar.$fieldExpr from $groupVar) )"
```

Comment:
Here you don't need to use subquery.
You can directly
select min(field)
from ...
where ...
group by ...